lib/faktory_worker/job.ex

```diff
@@ -43,11 +43,16 @@ defmodule FaktoryWorker.Job do
 
   @doc false
   def perform_async(payload, opts) do
-    pipeline_name = push_pipeline_name(opts)
+    opts
+    |> push_pipeline_name()
+    |> perform_async(payload, opts)
+  end
 
+  @doc false
+  def perform_async(pipeline_name, payload, _opts) do
     message = %Broadway.Message{
       acknowledger: {FaktoryWorker.PushPipeline.Acknowledger, :push_message, []},
-      data: payload
+      data: {pipeline_name, payload}
```

Comment:
It seems redundant to have to put the pipeline name in the data tuple as well as pass it as an argument to Broadway. Is this the only way to get the jobs re-enqueued because Broadway doesn't provide something built in?

Reply:
I haven't been able to find anything built in that allows a message to be put back into the queue it came from. The Broadway message also doesn't contain this info either and this was the simplest way I could find to get this working.